diff --git scripts/cl-update-checker scripts/cl-update-checker
index 67055ed..70d1a72 100644
--- scripts/cl-update-checker
+++ scripts/cl-update-checker
@@ -31,6 +31,7 @@ from calculate.update.update_info import UpdateInfo
 
 GUI_UPDATE_APP = DBUS_METHOD_APP % "update"
 
+MINUTE=60
 
 class SysTray(QtGui.QSystemTrayIcon):
     ICON = "/usr/share/pixmaps/calculate-console-update.png"
@@ -95,7 +96,7 @@ class DBusChecker(dbus.service.Object):
         self.parent.step()
 
 class CheckThread(QtGui.QMainWindow, UpdateInfo):
-    interval = 10
+    interval = MINUTE
 
     def __init__(self, bus):
         super(CheckThread, self).__init__()
