diff --git consolegui/application/ControlButton.py consolegui/application/ControlButton.py
index 3e42fe9..84b2511 100644
--- consolegui/application/ControlButton.py
+++ consolegui/application/ControlButton.py
@@ -289,7 +289,7 @@ class ControlButtonWgt(QtGui.QWidget):
         self.clear_proc_cache_button.setDisabled(True)
         sid = int(self.ClientObj.sid)
         res = self.ClientObj.client.service.clear_pid_cache(sid, self.pid)
-        if (hasattr(self._parent.ClientObj.sys_update_pid) and
+        if (hasattr(self._parent.ClientObj,"sys_update_pid") and
                     str(self.pid) == str(
                         self._parent.ClientObj._parent.sys_update_pid)):
             self._parent.ClientObj._parent.sys_update_pid = None
