diff --git consolegui/application/ControlButton.py consolegui/application/ControlButton.py
index 3e42fe9..ccd7576 100644
--- consolegui/application/ControlButton.py
+++ consolegui/application/ControlButton.py
@@ -146,6 +146,11 @@ class ControlButtonWgt(QtGui.QWidget):
         self.break_button = wgt
         self.hlayout.addWidget(wgt)
 
+    def close_break_button(self):
+        if self.break_button:
+            self.break_button.close()
+            self.break_button = None
+
     def _hide(self):
         if hasattr (self, 'button_widget'):
             self.hlayout.removeWidget(self.button_widget)
@@ -289,7 +294,7 @@ class ControlButtonWgt(QtGui.QWidget):
         self.clear_proc_cache_button.setDisabled(True)
         sid = int(self.ClientObj.sid)
         res = self.ClientObj.client.service.clear_pid_cache(sid, self.pid)
-        if (hasattr(self._parent.ClientObj.sys_update_pid) and
+        if (hasattr(self._parent.ClientObj,"sys_update_pid") and
                     str(self.pid) == str(
                         self._parent.ClientObj._parent.sys_update_pid)):
             self._parent.ClientObj._parent.sys_update_pid = None
diff --git consolegui/application/MainFrameResult.py consolegui/application/MainFrameResult.py
index 8a6498b..44e6c8f 100755
--- consolegui/application/MainFrameResult.py
+++ consolegui/application/MainFrameResult.py
@@ -884,13 +884,14 @@ class MainFrameRes(QtGui.QWidget):
                 self.send_password_thread.wait()
 
     def closeDialogs(self):
-        if hasattr(self,"message") and self.message:
+        if hasattr(self, "message") and self.message:
             self.message.close()
             self.message = None
-        if hasattr(self,"passwd") and self.passwd:
+        if hasattr(self, "passwd") and self.passwd:
             self.passwd.close()
             self.passwd = None
         self._parent.control_button.close_confirmation()
+        self._parent.control_button.close_break_button()
 
     def closeEvent(self, event):
         self.timer.stop()
@@ -1042,7 +1043,7 @@ class ButtonDialog(QtGui.QWidget):
             show_msg(_('Failed to send the message (password) to the server'))
             self.close()
             return 1
-        self._parent.show_result(result)
+        #self._parent.show_result(result)
         self.close()
 
 
diff --git consolegui/application/TrayIcon.py consolegui/application/TrayIcon.py
index 9dfe126..fca4379 100644
--- consolegui/application/TrayIcon.py
+++ consolegui/application/TrayIcon.py
@@ -21,7 +21,7 @@ import dbus
 
 from ReturnMessage import ReturnedMessage
 from calculate.consolegui.application.pid_information import client_pid_info
-from more import show_msg, get_sid
+from more import show_msg, get_sid, _print
 
 #class MySignal(QtCore.QObject):
 #    sig = QtCore.Signal(str, str)
@@ -48,7 +48,6 @@ class TrayUpdateSystem(QtCore.QThread):
         try:
             param_object = create_obj(self._parent.localhost_ClientObj.client,
                                       "update")
-            param_object.cl_update_precheck_set = True
             meth_result_temp = \
                 self._parent.localhost_ClientObj.client.service.update(
                     sid, param_object)
diff --git consolegui/application/more.py consolegui/application/more.py
index c50fbac..e82a91e 100755
--- consolegui/application/more.py
+++ consolegui/application/more.py
@@ -2301,7 +2301,7 @@ class TaskWidget(QtGui.QWidget):
         self.text.setMinimumHeight(self.text.sizeHint().height())
         self.text.setMinimumWidth(d_w)
         self._layout.addWidget(self.text)
-        self._layout.setContentsMargins(0, 4, 0, 4)
+        self._layout.setContentsMargins(0, 0, 0, 0)
         self._layout.setSpacing(8)
         self.status = False
 
