diff --git update/emerge_parser.py update/emerge_parser.py
index 597ba55..204e758 100644
--- update/emerge_parser.py
+++ update/emerge_parser.py
@@ -57,7 +57,8 @@ class CommandExecutor(object):
 
     def __init__(self, cmd, params, env=None, cwd=None, logfile=None):
         self.cwd = cwd
-        self.env = env
+        self.env = env or dict(os.environ)
+        self.env.update({'EINFO_QUIET':'NO'})
         self.cmd = cmd
         self.params = params
         self.child = None
diff --git update/update.py update/update.py
index ce66dd7..f9b1628 100644
--- update/update.py
+++ update/update.py
@@ -16,6 +16,7 @@
 
 import sys
 from os import path
+import os
 
 from calculate.lib.utils.tools import AddonError
 from calculate.lib.utils.colortext.palette import TextState
@@ -124,15 +125,21 @@ class Update:
                 if e.addon:
                     self.printWARNING(str(e.addon))
                 self.printWARNING(str(e))
-                self.printWARNING(_("Re-fetch {name} repository"
-                ).format(name=repname))
+                self.printWARNING(
+                    _("Re-fetch {name} repository").format(name=repname))
                 try:
+                    rpath_new = "%s_new" % rpath
+                    self._syncRepository(repname, url, rpath_new, revision,
+                                         branch)
                     removeDir(rpath)
-                except OSError as e:
+                    os.rename(rpath_new, rpath)
+                except OSError:
                     raise UpdateError(_("Permission denied to change "
                                         "{repname} repository").format(
                         repname=repname))
-        self._syncRepository(repname, url, rpath, revision, branch)
+        else:
+            if not self._syncRepository(repname, url, rpath, revision, branch):
+                return "skip"
         return True
 
     def syncLaymanRepository(self, repname):
diff --git update/utils/cl_update.py update/utils/cl_update.py
index 033a3bd..82d9247 100644
--- update/utils/cl_update.py
+++ update/utils/cl_update.py
@@ -133,8 +133,7 @@ class ClUpdateAction(Action):
          'group': __("Updating Calculate Utilities"),
          'tasks': [
              {'name': 'update:update_calculate_pkgs',
-              'message': __("Updating {0}").format(
-                  pkg_color("sys-apps/calculate-utilities")),
+              'message': __("Calculating dependencies"),
               'method': 'Update.emerge("-u","sys-apps/calculate-utilities")',
               'condition': need_upgrade('sys-apps/calculate-utilities$')
              },
