diff --git data/calculate data/calculate
index c7f60b9..60a24b0 100644
--- data/calculate
+++ data/calculate
@@ -134,10 +134,6 @@ start() {
 		touch /* /bin/* /sbin/*
 		udevadm trigger --action="add" --subsystem-match=net
 	fi
-	if [[ $roottype != "livecd" ]]
-	then
-		/usr/bin/eix-checkout &>/dev/null &
-	fi
 
 	/sbin/rc-update --update
 	return 0
diff --git install/distr.py install/distr.py
index ae94ddb..e6202e8 100644
--- install/distr.py
+++ install/distr.py
@@ -754,7 +754,7 @@ class PartitionDistributive(Distributive):
             mulipartDataNotBind = filter(lambda x: x[2]!="bind",
                                          self.getMultipartData())
             for dev, mountPoint, fileSystem, isFormat, partTable\
-                    in mulipartDataNotBind:
+                    in sorted(mulipartDataNotBind, key=lambda x:x[1]):
                 if fileSystem!="swap":
                     realMountPoint = pathJoin(mdirectory, mountPoint)
                     self._mountPartition(dev,realMountPoint,"-t %s"%fileSystem)
diff --git install/variables/system.py install/variables/system.py
index f810ea8..69e800e 100644
--- install/variables/system.py
+++ install/variables/system.py
@@ -86,8 +86,6 @@ class VariableOsFormatUse(ReadonlyVariable):
                 'value': "off"}
 
     def checkFunc(self,fs):
-        if fs == "btrfs" and self.Get('os_root_type') == 'livecd':
-            return "no"
         if "format" in FileSystemManager.supportFS[fs] and \
                path.exists(FileSystemManager.supportFS[fs]["format"]):
             return "yes"
