diff --git data/calculate data/calculate
index c7f60b9..60a24b0 100644
--- data/calculate
+++ data/calculate
@@ -134,10 +134,6 @@ start() {
 		touch /* /bin/* /sbin/*
 		udevadm trigger --action="add" --subsystem-match=net
 	fi
-	if [[ $roottype != "livecd" ]]
-	then
-		/usr/bin/eix-checkout &>/dev/null &
-	fi
 
 	/sbin/rc-update --update
 	return 0
diff --git install/distr.py install/distr.py
index ae94ddb..e6202e8 100644
--- install/distr.py
+++ install/distr.py
@@ -754,7 +754,7 @@ class PartitionDistributive(Distributive):
             mulipartDataNotBind = filter(lambda x: x[2]!="bind",
                                          self.getMultipartData())
             for dev, mountPoint, fileSystem, isFormat, partTable\
-                    in mulipartDataNotBind:
+                    in sorted(mulipartDataNotBind, key=lambda x:x[1]):
                 if fileSystem!="swap":
                     realMountPoint = pathJoin(mdirectory, mountPoint)
                     self._mountPartition(dev,realMountPoint,"-t %s"%fileSystem)
diff --git install/variables/net.py install/variables/net.py
index e9e400d..01b0032 100644
--- install/variables/net.py
+++ install/variables/net.py
@@ -790,3 +790,24 @@ class VariableOsInstallNetSettings(NetHelper,Variable):
 
     def choice(self):
         return [("","")]+self.Get('os_install_net_conf_available')
+
+class VariableOsInstallPxeIp(Variable):
+    """
+    IP адрес PXE сервера
+    """
+    type = "choice"
+    opt = ["--ip"]
+    metavalue = "IP"
+
+    def init(self):
+        self.label = _("PXE server IP")
+        self.help = "set IP address for PXE server"
+
+    def get(self):
+        for ipaddr in ifilter(None, self.Get('os_install_net_ip')):
+            return ipaddr
+        else:
+            return ""
+
+    def choice(self):
+        return filter(None, self.Get('os_install_net_ip'))
diff --git install/variables/system.py install/variables/system.py
index f810ea8..69e800e 100644
--- install/variables/system.py
+++ install/variables/system.py
@@ -86,8 +86,6 @@ class VariableOsFormatUse(ReadonlyVariable):
                 'value': "off"}
 
     def checkFunc(self,fs):
-        if fs == "btrfs" and self.Get('os_root_type') == 'livecd':
-            return "no"
         if "format" in FileSystemManager.supportFS[fs] and \
                path.exists(FileSystemManager.supportFS[fs]["format"]):
             return "yes"
diff --git install/wsdl_install.py install/wsdl_install.py
index 974b6a4..c9bc196 100644
--- install/wsdl_install.py
+++ install/wsdl_install.py
@@ -150,7 +150,8 @@ class Wsdl(WsdlBase):
         'groups':[
             lambda group:group(_("PXE install"),
                 normal=('cl_image_filename',),
-                expert=('os_install_pxe_path',),
+                expert=('os_install_pxe_path',
+                        'os_install_pxe_ip'),
                 next_label=_("Install"))]},
         # настройка аудио
         {
