diff --git install/variables/distr.py install/variables/distr.py
index 5c23c17..001b5cd 100644
--- install/variables/distr.py
+++ install/variables/distr.py
@@ -143,6 +143,7 @@ class DistroRepository(Linux):
 
         def listdistr(pathname):
             if path.exists(path.join(pathname,'etc/make.profile')) or \
+               path.exists(path.join(pathname,'etc/portage/make.profile')) or \
                path.exists(path.join(pathname,'livecd')) or \
                pathname.startswith('/dev/'):
                 return [pathname]
